Apache.Druid.Querying/Internal/Sections/SectionFactoryJsonMapper.cs

```diff
@@ -40,8 +40,23 @@ void MapCallParam(ElementFactoryCall.Parameter.Any param, string? callResultMemb
 
                     var (value, columnNames) = DruidExpression.Map(expression.Value, context.ColumnNames, context.DataSerializerOptions);
                     result.Add(expression.Name, value);
+                    
                     if (options.ExpressionColumnNamesKey is string existing)
-                        result.Add(existing, JsonSerializer.SerializeToNode(columnNames, context.QuerySerializerOptions));
+                    {
+                        if (!result.ContainsKey(existing))
+                        {
+                            result.Add(existing, JsonSerializer.SerializeToNode(columnNames, context.QuerySerializerOptions));
+                        }
+                        else
+                        {
+                            result.TryGetPropertyValue(existing, out JsonNode? node);
+                            
+                            string[] fields = node.Deserialize<string[]>()!;
+                            
+                            result.Remove(existing);
+                            result.Add(existing, JsonSerializer.SerializeToNode(fields.Concat(columnNames), context.QuerySerializerOptions));
+                        }
```

Comment:
Thanks for the pr :)

Nickpick, but we can use `JsonNode.Remove` to both check existance of, get value of and remove the json property resulting in a bit more concise code.

Plus, I expect if multiple druid expressions use the same column, we don't want to include it multiple times, so we should do a `Distinct` on the column names.
  
```suggestion
                        if (result.Remove(existing, out var node))
                        {
                            var existingColumnNames = node.Deserialize<IEnumerable<string>>()!;
                            columnNames = existingColumnNames.Concat(columnNames).Distinct().ToArray();
                        }

                        result.Add(existing, JsonSerializer.SerializeToNode(columnNames, context.QuerySerializerOptions));
                        }
```

Reply:
Thanks for the suggestion to include calling `Distinct` method on column names! The suggestion looks definitely cleaner :)